rclpy/rclpy/duration.py

```diff
@@ -49,6 +49,33 @@ def __str__(self) -> str:
             return 'Infinite'
         return f'{self.nanoseconds} nanoseconds'
 
+    def __add__(self, other: 'Duration') -> 'Duration':
+        if isinstance(other, Duration):
+            return Duration(nanoseconds=other.nanoseconds + self.nanoseconds)
+        return NotImplemented
+
+    def __sub__(self, other: 'Duration') -> 'Duration':
+        if isinstance(other, Duration):
+            return Duration(nanoseconds=self.nanoseconds - other.nanoseconds)
+        return NotImplemented
+
+    def __mul__(self, other: Union[int, float]) -> 'Duration':
+        if isinstance(other, int):
+            return Duration(nanoseconds=self.nanoseconds * other)
+        if isinstance(other, float):
+            if not math.isfinite(other):
+                if other == float('inf'):
+                    return Infinite
+                else:
+                    raise ValueError("Can't multiply duration with nan")
+            return Duration(nanoseconds=int(self.nanoseconds * other))
+        return NotImplemented
+
+    def __floordiv__(self, other: int) -> 'Duration':
+        if isinstance(other, int):
+            return Duration(nanoseconds=self.nanoseconds // other)
+        return NotImplemented
+
```

Comment:
I am not sure if we need to support division, instead user can do `Duration * 0.5` as multiplication? (note that rclcpp does not have this operator.)

Reply:
I can open a PR in RCLCPP for this if we need. My take is that if someone were to want to do division by 3 its much better to say `//3` than `*0.333`.

Reply:
i think that is okay to have them, thanks. i am happy to review rclcpp too.

probably we need to add,

- `__truediv__` and roundup the result? to support `Duration / 3`?
- catch `ZeroDivisionError` that is missing here.

Reply:
Division always makes me somewhat uneasy.  One reason for that is https://en.cppreference.com/w/cpp/chrono/duration/operator_arith4, particularly API (6).

At the end of the day, I'm fine with implementing division, as long as it is division by a scalar (not by another Duration).  If we are going to do this, then I agree that we should do it both here in rclpy and rclcpp.

Reply:
> __truediv__ and roundup the result? to support Duration / 3?

 I don't think we should implement `__truediv__`, in fact I think that this is exactly the use case for `__floordiv__` is as it guarantees that the output is an integer and makes the programmer aware that rounding is occurring. 
 
 > catch ZeroDivisionError that is missing here
 
 Im curious what I would do with the exception, do we bubble a different exception up? I dont know if silently failing is a good idea.
 
 >  I'm fine with implementing division, as long as it is division by a scalar (not by another Duration)
 
Also curious about the rationale for this. There is a narrow use case for division by duration that I can think of (in fact something that I'm actively pursuing): If we are doing some form of discretized time based planning then division by duration is kind of nice for us to be able to set a size.

I'm open to removing division as well or separating this out into a seperate PR along with the relevant work in RCLCPP.

Reply:
> Also curious about the rationale for this. There is a narrow use case for division by duration that I can think of (in fact something that I'm actively pursuing): If we are doing some form of discretized time based planning then division by duration is kind of nice for us to be able to set a size.

See API (6) in `std::chono::duration` in https://en.cppreference.com/w/cpp/chrono/duration/operator_arith4 .  It doesn't return the same "type" as e.g. API (5), or API (3) and (4) for multiplication.  That makes me think that there are subtleties there that we need to worry about, and unless we have an immediate use case for it, I'd rather not go down that particular rabbit hole.

Reply:
I'm removing division from here. I'll probably open a seperate PR for it later so that we can move this PR along.